drivers/LVHDSR.py

```diff
@@ -262,6 +262,9 @@ def updateSRMetadata(self, allocation):
             for vdi in self.session.xenapi.SR.get_VDIs(self.sr_ref):
                 vdi_uuid = self.session.xenapi.VDI.get_uuid(vdi)
 
+                if self.session.xenapi.VDI.get_sm_config(vdi).get('vdi_type') is None:
+                    raise xs_errors.XenError('MetadataError', opterr=f"Missing vdi_type for vdi {vdi_uuid}")
```

Comment:
Empty string is also an invalid case. Normally it should not be returned here, but it's probably safer to do:
```suggestion
                if not self.session.xenapi.VDI.get_sm_config(vdi).get('vdi_type'):
                    raise xs_errors.XenError('MetadataError', opterr=f"Missing `vdi_type` for VDI {vdi_uuid}")
```